rcl/test/rcl/test_graph.cpp

```diff
@@ -205,9 +246,186 @@ TEST_F(
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
 
+/* Test the rcl_get_publisher_names_and_types_by_node function.
+ *
+ * This does not test content of the response.
+ */
+TEST_F(
+  CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publisher_names_and_types_by_node
+) {
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_node_t zero_node = rcl_get_zero_initialized_node();
+  const char * node_name = "/test_rcl_get_publisher_names_and_types_by_node";
+  const char * node_ns = "/test/namespace";
+  rcl_names_and_types_t nat = rcl_get_zero_initialized_names_and_types();
+  // invalid node
+  ret = rcl_get_publisher_names_and_types_by_node(
+    nullptr, &allocator, false, node_name, node_ns, &nat);
+  EXPECT_EQ(RCL_RET_NODE_INVALID, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+  ret = rcl_get_publisher_names_and_types_by_node(
+    &zero_node, &allocator, false, node_name, node_ns, &nat);
+  EXPECT_EQ(RCL_RET_NODE_INVALID, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+  ret = rcl_get_publisher_names_and_types_by_node(
+    this->old_node_ptr, &allocator, false, node_name, node_ns, &nat);
+  EXPECT_EQ(RCL_RET_NODE_INVALID, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+  // invalid allocator
+  ret = rcl_get_publisher_names_and_types_by_node(
+    this->node_ptr, nullptr, false, node_name, node_ns, &nat);
```

Comment:
@jacobperron here and elsewhere, consider also testing with a non-null but invalid allocator e.g. a zero initialized one as returned by [`rcutils_get_zero_initialized_allocator()`](https://github.com/ros2/rcutils/blob/master/include/rcutils/allocator.h#L84).

Reply:
Done 039e5fe